### PR TITLE
docs: update release process - WIP

### LINKED ISSUE #2357: Update `RELEASING.md` to include comprehensive release cycle instructions
# Problem

RELEASING.md is not accurate or sufficiently clear. 

# Closing criteria

Update RELEASING.md to include comprehensive release cycle instructions

### PR DESCRIPTION
**Pull request by mpoke**
This is a draft of the release process, which is still WIP. See discussion bellow for more details. 

Closes https://github.com/cosmos/gaia/issues/2357

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    - [Test building artifacts](#test-building-artifacts)
+    - [Release notes](#release-notes)
+
+
+This document outlines the release process for Cosmos Hub (Gaia).
+
+Gaia follows [semantic versioning](https://semver.org), but with the following deviations to account for state-machine breaking changes: 
+- Changes that requires upgrade trough governance will result in an increase of the major version X (X.y.z). Note that most likely this entails state-machine breaking changes.
+- Changes that are state-machine breaking but do not require upgrade through governance (e.g., emergency upgrades) will result in an increase of the minor version Y (x.Y.z | x > 0).
+- Changes that are not state-machine breaking will result in an increase of the patch version Z (x.x.Z | x > 0).
```
Comment (mpoke) [inline RELEASE_PROCESS.md]: This is similar to how we were already using versioning, but now we have it as a guideline.

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+> Note: Generally, PRs should target `main` (expect PRs open via the Github mergify integration). 
+
+* Once the team feels that `main` is feature complete, we create a `release/vY` branch (going forward known a release branch), 
+  where `Y` is the version number, with the patch part substituted to `x` (eg: 8.0.x). 
+  * Update the [GitHub mergify integration](./.mergify.yml) by adding instructions for automatically backporting commits from `main` to the `release/vY` using the `A:backport/vY` label.
+  * **PRs targeting directly a release branch can be merged _only_ when exceptional circumstances arise**.
+* In the release branch prepare a new version section in the `CHANGELOG.md`
+    * All links must point to their respective pull request.
+    * The `CHANGELOG.md` must contain only the changes of that specific released version.
```
Comment (mpoke) [inline RELEASE_PROCESS.md:51]: TODO: Currently, the changelog contains all the releases, including release candidates. We should only keep release candidates before we cut a final release. Then, the changelog in a release branch should contain only changes to that release branch. See https://github.com/cosmos/cosmos-sdk/blob/release/v0.46.x/CHANGELOG.md

File: RELEASE_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+* Once the team feels that `main` is feature complete, we create a `release/vY` branch (going forward known a release branch), 
+  where `Y` is the version number, with the patch part substituted to `x` (eg: 8.0.x). 
+  * Update the [GitHub mergify integration](./.mergify.yml) by adding instructions for automatically backporting commits from `main` to the `release/vY` using the `A:backport/vY` label.
+  * **PRs targeting directly a release branch can be merged _only_ when exceptional circumstances arise**.
+* In the release branch prepare a new version section in the `CHANGELOG.md`
+    * All links must point to their respective pull request.
+    * The `CHANGELOG.md` must contain only the changes of that specific released version. 
+      All other changelog entries must be deleted and linked to the `main` branch changelog ([example](TBA)).
+    * Create release notes, in `RELEASE_NOTES.md`, highlighting the new features and changes in the version.
```
Comment (mpoke) [inline RELEASE_PROCESS.md:53]: TODO: Set up goreleaser. See https://github.com/cosmos/cosmos-sdk/blob/a392bae9b88f491d02cfa1fb5ea89e8d605ef840/.github/workflows/release.yml#L26 and https://github.com/cosmos/cosmos-sdk/blob/a392bae9b88f491d02cfa1fb5ea89e8d605ef840/.goreleaser.yml#L2

File: RELEASE_PROCESS.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+  where `Y` is the version number, with the patch part substituted to `x` (eg: 8.0.x). 
+  * Update the [GitHub mergify integration](./.mergify.yml) by adding instructions for automatically backporting commits from `main` to the `release/vY` using the `A:backport/vY` label.
+  * **PRs targeting directly a release branch can be merged _only_ when exceptional circumstances arise**.
+* In the release branch prepare a new version section in the `CHANGELOG.md`
+    * All links must point to their respective pull request.
+    * The `CHANGELOG.md` must contain only the changes of that specific released version. 
+      All other changelog entries must be deleted and linked to the `main` branch changelog ([example](TBA)).
+    * Create release notes, in `RELEASE_NOTES.md`, highlighting the new features and changes in the version. 
+      This is needed so the bot knows which entries to add to the release page on GitHub.
+    * Additionally verify that the `UPGRADING.md` file is up to date and contains all the necessary information for upgrading to the new version.
```
Comment (mpoke) [inline RELEASE_PROCESS.md:55]: TODO: This may be a good way to have upgrading info per release.

File: RELEASE_PROCESS.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+  * Add more tests.
+  * Fix bugs as they are discovered.
+* After the team feels that the release branch works fine (i.e., has `~90%` chance of reaching mainnet), we cut a release candidate.
+  * Create a new annotated git tag for a release candidate in the release branch (follow the [Tagging Procedure](#tagging-procedure)).
+  * The release verification on public testnets must pass. 
+  * When bugs are found, create a PR for `main`, and backport fixes to the release branch.
+  * Create new release candidate tags after bugs are fixed.
+* After the team feels the release candidate is mainnet ready, create a full release:
+  * Update `CHANGELOG.md`.
+  * Run `gofumpt -w -l .` to format the code.
```
Comment (mpoke) [inline RELEASE_PROCESS.md:67]: Not sure whether we already do this.

File: RELEASE_PROCESS.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+- Test and benchmarking suites, ensuring that fixes are sound and there are no performance regressions
+- Library updates including point releases for core libraries such as IBC-Go, Cosmos SDK, Tendermint and other dependencies
+- General maintenance improvements, that are deemed necessary by the stewarding team, that help align different releases and reduce the workload on the stewarding team
+- Security fixes
+
+Issues that are likely excluded, are any issues that impact operating a block producing network.
+
+---------
+
+# Old Release Process
```
Comment (mpoke) [inline RELEASE_PROCESS.md:146]: TODO: Figure out what needs to be kept from the previous release process docs.

File: RELEASE_PROCESS.md (hunk 5 above)
Comment (mmulji-ic) [reply RELEASE_PROCESS.md:67]: Maybe a git hook before committing?

File: RELEASE_PROCESS.md (hunk 2 above)
Comment (mpoke) [reply RELEASE_PROCESS.md:51]: https://github.com/cosmos/gaia/pull/2295 Fixes this todo

**Comment by mpoke** [comment]
Closing for now as the work is paused.

**Pull request closed without merging**